Formula/m/mlx.rb

```diff
@@ -0,0 +1,72 @@
+class Mlx < Formula
+  desc "Array framework for Apple silicon"
+  homepage "https://github.com/ml-explore/mlx"
+  url "https://github.com/ml-explore/mlx/archive/refs/tags/v0.4.0.tar.gz"
+  sha256 "36ef8aaec328ae93869f9e2906aba3bf1cfee6e4bc134215e0294b9a24a889f1"
+  license "MIT"
+  head "https://github.com/ml-explore/mlx.git", branch: "main"
+
+  depends_on "cmake" => :build
+  depends_on "pybind11" => :build
```

Comment:
```suggestion
  depends_on "pybind11" => :build
  depends_on "python-setuptools" => :build
```

Reply:
I wasn't sure which direction to go either, but I can certainly try that, yes.